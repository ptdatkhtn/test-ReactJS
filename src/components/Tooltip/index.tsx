import Tooltip from './styles'

export default Tooltip
