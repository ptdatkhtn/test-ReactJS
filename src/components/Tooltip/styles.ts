import { Tooltip } from 'antd'
import styled from 'styled-components'

export default styled(Tooltip)``
