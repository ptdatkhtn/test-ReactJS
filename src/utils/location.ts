export const degreeToRadian = (degree : number): number => {
  return degree * Math.PI / 180
}

export const calculateDistanceInKm = (lat1: number, lon1: number, lat2: number, lon2: number): number => {
  const earthRadiusKm = 6371

  // have to convert degree unit to radian unit before the calculation
  const dLat = degreeToRadian(lat2 - lat1)
  const dLon = degreeToRadian(lon2 - lon1)

  lat1 = degreeToRadian(lat1)
  lat2 = degreeToRadian(lat2)

  const a = Math.sin(dLat / 2) * Math.sin(dLat / 2) + Math.sin(dLon / 2) * Math.sin(dLon / 2) * Math.cos(lat1) * Math.cos(lat2)
  const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a))
  return earthRadiusKm * c
}