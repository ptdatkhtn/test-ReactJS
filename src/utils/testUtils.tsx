import { render as rtlRender } from '@testing-library/react'
import React from 'react'
import { ThemeProvider } from 'styled-components'
import theme from 'themes'

const render = (ui: JSX.Element, { ...renderOptions } = {}) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return (
      <ThemeProvider theme={theme}>
        {children}
      </ThemeProvider>
    )
  }
  return rtlRender(ui, { wrapper: Wrapper as React.FC, ...renderOptions })
}

const fixMatchMedia = () => {
  window.matchMedia = window.matchMedia || function() {
    return {
        matches: false,
        addListener: function() {},
        removeListener: function() {}
    }
  }
}

export * from '@testing-library/react'
// override render method from @testing-library/react
export { render, fixMatchMedia }
