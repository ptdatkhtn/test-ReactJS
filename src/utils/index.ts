import { calculateDistanceInKm } from './location'

export { calculateDistanceInKm }