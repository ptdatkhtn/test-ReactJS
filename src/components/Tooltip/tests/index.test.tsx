import React from 'react'
import 'jest-styled-components'
import Tooltip from '..'
import { render } from 'utils/testUtils'

describe('Tooltip', () => {
  it('should render without error', () => {
    const { container } = render(
      <Tooltip placement='top' title='tooltip'/>
    )
    expect(container).toMatchSnapshot()
  })
})