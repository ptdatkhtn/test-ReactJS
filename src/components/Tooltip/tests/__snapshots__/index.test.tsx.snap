// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should render without error 1`] = `
<div>
  <span
    class=""
  />
</div>
`;
